### PR TITLE
cache theme lookup and add ability to filter them

### PR DESCRIPTION
**Pull request by nkryptic**
The basis for this PR was that we needed to remove all theme choices besides those that were "approved".  Without something like this, I had to hook into the `m_chart_settings_template` and write a "wrapper" template that captured the output of rendering the actual settings template.  Then, I removed the unwanted theme options with `preg_replace`.  Not very pretty.

**What this PR does:**
Caches the themes found during the `get_themes` call and adds a filter hook on the themes, providing the ability to add/remove programmatically.

### CODE REVIEW

File: components/class-m-chart-highcharts.php [hunk 1]
```diff
@@ -620,12 +620,22 @@ public function fix_null_values( $value ) {
 	 * @return array an array of themes
 	 */
 	public function get_themes() {
+		$cache_key = 'm-chart-highcharts-available-themes';
+
+		if ( $themes = wp_cache_get( $cache_key ) ) {
```
Comment (methnen) [inline components/class-m-chart-highcharts.php:625]: Take a look here for how I'd prefer the caching key stuff handled (i.e. use the group parameter for the name spacing):

https://github.com/methnen/m-chart-highcharts-library/blob/06b0177d2fb1fa5ceab70210be77e3c9a33e4d70/components/class-m-chart-highcharts.php#L126

**Review by methnen — COMMENTED** [review]
Looks great except for the minor note on the cache key stuff.

I also don't see how this cache ever gets refreshed without the object cache just working it's way through whatever caching mechanism is being used. In some cases with a really aggressive and long lasting object cache that could be problematic.

Perhaps add some code to delete the cache when someone hits the M Chart Settings page?

**Comment by nkryptic** [comment]
I'll make the recommended changes.  As to the cache refreshing, I've only ever used the built-in object cache which refreshes on each page load, so thanks for the tip.  I think refreshing the cache on the settings page is a must.  Do you think it acceptable to not refresh the cache on the new/edit chart page?

**Comment by nkryptic** [comment]
I've made the changes.  Here's what would be added to the [Action and filter hooks](https://github.com/methnen/m-chart/wiki/Action-and-filter-hooks) page in the m-chart wiki: [m-chart-wiki-flexible-themes-pr.md](https://github.com/methnen/m-chart-highcharts-library/files/8196000/m-chart-wiki-flexible-themes-pr.md)

**Comment by nkryptic** [comment]
I've merged the refresh for the new/edit chart page and added the relevant hook to the wiki page.

**Pull request merged**